website/blog/2022-10-31-simple-1.mdx

```diff
@@ -0,0 +1,108 @@
+---
+slug: simple-functions-1
+title: "Simple Functions: Introduction and Basic Optimizations"
+authors: [lsakka]
+tags: [tech-blog, simple-functions]
+---
+
+_This blogpost is part of a series of blog posts that discuss different features and optimizations of the simple function interface in Velox_.
+
+
+## Introduction to Simple Functions
+
+Scalar functions are one of the most used extension points in Velox. Since Velox is a vectorized engine, by nature functions are <a href="https://facebookincubator.github.io/velox/develop/scalar-functions.html#vector-functions">"vector functions"</a> that consume Velox vectors (batches of data) and produce vectors. Velox allows users to write functions as vector functions or as single-row operations "simple functions" that are converted to vector functions using template expansion through <a href="https://github.com/facebookincubator/velox/blob/main/velox/expression/SimpleFunctionAdapter.h">SimpleFunctionAdapter</a>.
+
+<figure>
+<img src="/img/simple1_1.png"/>
+</figure>
+
+Writing functions as vector functions directly gives the user `complete` control over the function implementations and optimizations, however it comes with some cost that can be summarized in three points:
+
+- **Complexity** : Requires an understanding of Velox vectorized data representation and encodings, which requires additional work for our customers, specially those without DB background. And writing optimized vector functions requires even deeper understanding.
+- **Repetition** : Involves repeated efforts and code; in each function vectors are read in the same way, same set of optimizations are applied ... etc.
```

Comment:
this reads a little vague. Can we be more specific, or maybe give one concrete example?